### PR TITLE
fix(billing): run the 3DS confirm exactly once per payment intent

### PR DESCRIPTION
**Pull request by baktun14**
## Why

Adding credits with a **new** card that requires 3D Secure was glitchy: the challenge opened, closed itself, an error toast ("We are unable to authenticate your payment method…") appeared, and then a **second** challenge opened. If the user completed that second challenge, Stripe charged the card and the webhook credited the wallet — while the UI showed a failure. Already-saved 3DS cards were unaffected (exactly one challenge).

Root cause (two parts):

- **Re-entrant confirm** — `ThreeDSecureForm` fired `stripe.confirmPayment` inside an effect whose deps included callbacks chained to unstable parent props, and the effect **cleanup reset the in-progress guard**. So any mid-challenge parent re-render re-ran the effect and fired a second `confirmPayment` on the same PaymentIntent: Stripe cancelled the first challenge (popup closes), its promise resolved with an auth error (toast + failure), and the second confirm opened a fresh challenge.
- **The new-card-only trigger** — `confirmPayment.onSuccess` invalidated the payment-methods query even on the `202 { requiresAction: true }` response. For a new card the refetched list changed (the card was just saved via `confirmSetup`), producing a new reference that re-rendered the form mid-challenge. For a saved card the list was deep-equal → structural sharing kept the same reference → no re-render → single clean challenge.

Backend is sound and untouched (202 with clientSecret, attempt-key idempotency resumes the same PaymentIntent, crediting is webhook-only).

No linked Linear issue.

## What

- **`ThreeDSecureModal`** — the 3DS confirmation now runs **exactly once per PaymentIntent**: a run-once latch that no cleanup resets (safe because `<Elements key={clientSecret}>` plus conditional mounting remount the form for every genuinely new challenge), and a latest-callbacks ref so parent re-renders can neither re-fire nor abort the in-flight confirm while the freshest `onSuccess`/`onError` still run on resolution. Result routing (success / declined / unexpected status / missing intent / thrown error) is unchanged. Added a `DEPENDENCIES` export for DI-based testing and dropped stray `console.*` calls.
- **`usePaymentQueries`** — `confirmPayment.onSuccess` no longer invalidates the payment-methods query while a payment still `requiresAction` (a 202 settles nothing; the 3DS-success path invalidates via `validatePaymentMethodAfter3DS.onSuccess` and the failure path via the form's own handler). The transactions invalidation stays unconditional.

Note: a brand-new 3DS card still sees **two sequential, clean** challenges (one saving the card, one for the charge) — inherent to the current save-then-charge design and accepted here; this PR only removes the glitch (self-closing challenge + false error + duplicate popup).

### Tests

- New `ThreeDSecureModal.spec.tsx` (TDD): the key regression asserts `confirmPayment` fires **exactly once** across parent re-renders with fresh callback identities (failed with 3 calls before the fix), plus success routes to the latest `onSuccess` after the delay, all failure routings, unmount safety, and the stripe-unavailable / empty-client-secret guards.
- Extended `usePaymentQueries.spec.tsx`: `requiresAction` response → payment methods **not** invalidated (transactions still are); settled response → both invalidated (asserted via `QueryKeys.*` helpers).

### Manual QA (Stripe test mode, card `4000 0027 6000 3184`)

- New-card top-up → one setup challenge + one payment challenge, no error toast, no form reshuffle behind the popup, credits arrive.
- Same flow, click FAIL → single error surfaced, no second popup, retry works.
- Saved-card top-up → exactly one challenge.
- Onboarding trial start with a 3DS card → one challenge, wallet created.

## Summary by CodeRabbit

* **New Features**
  * Enhanced 3D Secure payment authentication with clearer handling for success, failures, declines, unknown states, and user-facing retry messaging.
* **Bug Fixes**
  * Prevented duplicate Stripe confirmations and ensured only the latest success/error callbacks are invoked, including during unmounts mid-challenge.
  * Improved resilience when Stripe/auth data is unavailable, and improved error messages for unexpected or malformed responses.
  * Refined data refresh after confirmation: payment methods are invalidated only when no additional 3DS action is required.
* **Tests**
  * Added/updated coverage for confirmation logic, callback timing, unmount safety, error edge cases, and query invalidation behavior.

### CODE REVIEW

File: apps/deploy-web/src/components/shared/PaymentMethodForm/ThreeDSecureModal.tsx [hunk 1]
```diff
@@ -51,107 +60,70 @@ const ThreeDSecureForm: React.FC<Omit<ThreeDSecureModalProps, "isOpen" | "onClos
...
-        const errorMessage = "Your payment method was declined. Please try a different card.";
-        handleAuthenticationFailure(errorMessage);
-      } else {
-        console.error("3D Secure authentication failed, unexpected status:", paymentIntent.status);
-        const errorText = `Authentication failed. Status: ${paymentIntent.status || "unknown"}`;
-        handleAuthenticationFailure(errorText);
+      function succeed() {
+        setStatus("succeeded");
+        setTimeout(() => callbacksRef.current.onSuccess(), SUCCESS_DELAY);
       }
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/deploy-web/src/components/shared/PaymentMethodForm/ThreeDSecureModal.tsx:102]: Good catch — fixed in 0d33007b1. The `SUCCESS_DELAY` timer is now stored in a `successTimerRef` and cleared in an unmount cleanup effect, with a test covering the unmount-during-success-delay path.

**Review by ygrishajev — APPROVED** [review]
